lib/qiita/markdown/filters/user_input_sanitizer.rb

```diff
@@ -2,30 +2,90 @@ module Qiita
   module Markdown
     module Filters
       # Sanitizes raw markdown input if :strict context is given.
-      class UserInputSanitizer < HTML::Pipeline::TextFilter
+      class UserInputSanitizer < HTML::Pipeline::Filter
+        class AttributeTransformer
+          FILTERS = {
+            "a" => {
+              "class" => %w[autolink],
+              "rel" => %w[footnote url],
+              "rev" => %w[footnote],
+            },
+            "sup" => {
+              "id" => /^fnref\d+$/,
```

Comment:
We always want to use `\A` instead of `^`, and `\z` instead of `$` when possible.